package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
```

Comment:
Is there a reliable way to detect a NPT system? I could detect boxes that are `[0, 0, 0, 90, 90, 90]` but I am not sure it is much better than copying the box.